### ISSUE TITLE
[Qt] Can we please add icons_rc.py to this repository?

### ISSUE THREAD

**Issue by cculianu** [Electron-Cash/Electron-Cash#1088]
**TL;DR:** Can we please add `gui/qt/icons_rc.py` to this github repository and call it a day?

**Longer exposition:** Guys, I can't think of a single reason for this `icons_rc.py` file to *not* be a part of the repository.  All it does it create a headache every time we change the icons, cause possible bad UX for users using stale versions of the file (read: missing icons), and overcomplicate the build scripts.

It also is a minor hassle for Linux users cloning the repo and running from source as they get a lovely error message and have to run pyrcc3 themselves.  (They also then get the surprises of possibly-stale-icons if they `git pull` and we have since updated the icons..)

I am trying to imagine the original rationale for this in the first place and I suspect the original programmers did this because icons_rc format for Qt is unstable across Python versions (there was a time when Electrum supported both Python2 and Python3, correct?). So that's why you had to build it yourself for your Python version.. that's the only reason I can imagine for this (and I could even be wrong about it).

But if that was the reason: Those days are gone, and this `icons_rc.py`-living-in-a-separate-repo business is just a hassle that helps noone, least of all us as maintainers.

So -- I propose we add `icons_rc.py` -- which is *stable* for Python3 and various versions of PyQt5 -- and *call it a day*.

**Pros:**

- Less hassle
- Easier on users running from source
- Less confusion

**Cons:**

- Can't think of a single one.  Can you?

**Steps needed to accomplish this**: 

1. `git add gui/qt/icons_rc.py`
2. Modify build scripts for Windows and Mac to no longer look for or check out `Electron-Cash/electrum-icons`.
3. Profit.

---

@fyookball  @markblundeberg  Please comment on this and/or let me know your thoughts and/or give me the green light to pull the plug on this holdover from a bygone era.

Thanks,

-Calin

**Comment by cculianu** [comment]
Thanks for the counter argument.

I take issue with the use of the word “compiled” though. :)

I disagree that technically there is anything *compiled* about this file, though, is my counterpoint.

It’s just binary icon files (which are already in the repo by the way), transmogrified to be python arrays so they live in memory as such when the app is launched.

To me this is no worse or better than having the icon files live in the repository.

The use of the word “compiled” implies they can change across platforms or versions of things and thus that they do not belong in a repository. I’d argue that’s not the case here as it’s just python data, the format for which is relatively stable (but I concede not entirely constant).

**Comment by markblundeberg** [comment]
Yeah, I think in principle it's fine, it just maybe doesn't go in the right direction if we will end up adding more 'compiled' things like Electrum has.

**Comment by cculianu** [comment]
Sorry I edited my comment ... while you were replying. I rearranged some words.

So no chance I can get you on board with my new revised masterpiece comment? :)

**Comment by cculianu** [comment]
Is if you consider it a cardinal sin against all that is holy we won’t do it. Grumble grumble...

**Comment by markblundeberg** [comment]
No, I don't think it's a big problem. If we do it, it's fine. :-)

Actually also, I find the electrum-icons and electrum-locale repository thing to be even more hideous. So any move away from that can only be an improvement. :-)

**Comment by cculianu** [comment]
Ah ok, I must have misunderstood.  I thought you were just being polite but you thought it was evil.

Yeah electrum-locale was in the spirit of deterministic builds (which, if I am not mistaken, we actually do have for Window but not Mac) -- so I can sort of see the point of that once (because the locale files are super sensitive to lots of things so not having to rebuild them each time but rather pull them from a known source makes determinism easier to accomplish).

My only gripe with electrum-icons and/or having to manually (re)build the icons_rc.py file is a purely pragmatic one.  Whenever we update the icons there is the inevitable user(s) that don't realize we did so, and there's no effective way to tell them we did -- so they end up getting old/stale/missing icons when they fire up EC and who knows if they will ever realize they need to rebuild the file.  That "loose end" as it were for users is the only thing that really irks me about icons_rc.py not living within the repo.

And.. since icons_rc.py is very much not very sensitive to the exact layout of the code or anything like that and should change once per decade if we don't change icons (unlike the locale files that have code line numbers in them and are huge) -- I see no harm in adding it.

OK.. I'm beating a dead horse here.

I'll sit on this for a day or two and the add icons_rc.py unless Jonald yells at me not to...

Thanks. :)

**Comment by fyookball** [comment]
I don't understand what electrum is doing now but as far as the overall philosophy, i'm ok to include it; i think the bloat is very minor and the convenience may be worth it.

**Comment by cculianu** [comment]
Yay! Thanks man.  Yeah you'll see.  It's super easy, barely an inconvenience! :D

**Comment by marceloneil** [comment]
I am for this. I think we should take adding compiled elements on a case by case basis, but in this case I think it makes sense

**Comment by cculianu** [comment]
Ha.. hey! Marcel! You're back!

Agreed.  Case by case is good. And glad you agree in this case it makes sense.. 🗡

**Comment by EchterAgo** [comment]
A drawback of this change is that it is less clear how to actually change the icons. How to generate `icons.py` using `pyrcc5` should still be documented. I just spent quite a short while searching for the command as I remembered using it before. Grepping the source for `pyrcc` or `icons.py` yields no results.

**Comment by EchterAgo** [comment]
```
pyrcc5 -o gui/qt/icons.py icons.qrc
```

**Comment by cculianu** [comment]
:)
